src/Columns/ColumnTuple.cpp

```diff
@@ -169,7 +169,10 @@ void ColumnTuple::insertDefault()
 void ColumnTuple::popBack(size_t n)
 {
     for (auto & column : columns)
-        column->popBack(n);
+    {
+        if (column->size() >= n)
+            column->popBack(n);
+    }
```

Comment:
But how is it possible to have columns with different sizes?
Maybe better to throw logical error in this case, because it's unexpected

Reply:
Probably in `SerializationTuple` we don't process exceptions during columns parsing correctly